.github/workflows/validate.yml

```
@@ -0,0 +1,239 @@
+        run: |
+          sudo apt-get install clang-3.9 -qyy
+          source ./util/github/common.sh
+          install_linux_deps
+
+      - name: Build
+        run: |
+          mkdir cmakebuild
+          cd cmakebuild
+          cmake -DCMAKE_BUILD_TYPE=Debug -DRUN_IN_PLACE=TRUE -DENABLE_GETTEXT=TRUE -DBUILD_SERVER=TRUE ${CMAKE_FLAGS} ..
+          make -j2
+        env:
+          CMAKE_FLAGS: "-DCMAKE_C_COMPILER=clang-3.9 -DCMAKE_CXX_COMPILER=clang++-3.9"
+
+      - name: Test
+        run: |
+          ./bin/minetest --run-unittests
+
+  # This is the current clang version
+  clang_9:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install compiler
+        run: |
+          sudo add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+          sudo apt-get install clang-9 -qyy
+          source ./util/github/common.sh
+          install_linux_deps
+
+      - name: Build
+        run: |
+          mkdir cmakebuild
+          cd cmakebuild
+          cmake -DCMAKE_BUILD_TYPE=Debug -DRUN_IN_PLACE=TRUE -DENABLE_GETTEXT=TRUE -DBUILD_SERVER=TRUE ${CMAKE_FLAGS} ..
+          make -j2
+        env:
+          CMAKE_FLAGS: "-DCMAKE_C_COMPILER=clang-9 -DCMAKE_CXX_COMPILER=clang++"
+
+      - name: Test
+        run: |
+          ./bin/minetest --run-unittests
+          
+  # Some builds doesn't require freetype, ensure it compiled properly
+  clang_9_no_freetype:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install compiler
+        run: |
+          sudo add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+          sudo apt-get install clang-9 -qyy
+          source ./util/github/common.sh
+          install_linux_deps
+
+      - name: Build
+        run: |
+          mkdir cmakebuild
+          cd cmakebuild
+          cmake -DCMAKE_BUILD_TYPE=Debug -DRUN_IN_PLACE=TRUE -DENABLE_GETTEXT=TRUE -DBUILD_SERVER=TRUE -DENABLE_FREETYPE=0 ${CMAKE_FLAGS} ..
+          make -j2
+        env:
+          CMAKE_FLAGS: "-DCMAKE_C_COMPILER=clang-9 -DCMAKE_CXX_COMPILER=clang++"
+
+      - name: Test
+        run: |
+          ./bin/minetest --run-unittests
+
+  # Run tests with valgrind to verify memleaks on the tested code
+  clang_9_valgrind:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install compiler
+        run: |
+          sudo add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+          sudo apt-get install clang-9 valgrind -qyy
+          source ./util/github/common.sh
+          install_linux_deps
+
+      - name: Build
+        run: |
+          mkdir cmakebuild
+          cd cmakebuild
+          cmake -DCMAKE_BUILD_TYPE=Debug -DRUN_IN_PLACE=TRUE -DENABLE_GETTEXT=TRUE -DBUILD_SERVER=TRUE ${CMAKE_FLAGS} ..
+          make -j2
+        env:
+          CMAKE_FLAGS: "-DCMAKE_C_COMPILER=clang-9 -DCMAKE_CXX_COMPILER=clang++"
```

same issue in other places
```suggestion
          CMAKE_FLAGS: "-DCMAKE_C_COMPILER=clang-9 -DCMAKE_CXX_COMPILER=clang++-9"
```